app/src/main/java/com/github/se/polyfit/ui/screen/IngredientScreen.kt

```diff
@@ -0,0 +1,214 @@
+package com.github.se.polyfit.ui.screen
+
+import android.util.Log
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ExperimentalLayoutApi
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Add
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.compose.rememberNavController
+import com.github.se.polyfit.ui.components.GradientButton
+import com.github.se.polyfit.ui.navigation.Navigation
+import com.github.se.polyfit.ui.theme.PrimaryPurple
+
+// TODO: Replace with real data class
+data class Ingredient(
+    val name: String,
+    val quantity: Int,
+    val unit: String,
+    val probability: Float = 1.0f,
+) {}
+
+@Composable
+fun IngredientScreen(
+    navigation: Navigation,
+    initialIngredients: List<Ingredient>,
+    initialPotentialIngredients: List<Ingredient>,
+    // TODO:  ingredientsViewModel: IngredientsViewModel = IngredientsViewModel(),
+) {
+
+  val ingredients = remember { mutableStateListOf(*initialIngredients.toTypedArray()) }
+  val potentialIngredients = remember {
+    mutableStateListOf(*initialPotentialIngredients.toTypedArray())
+  }
+
+  Scaffold(topBar = { TopBar(navigation) }, bottomBar = { BottomBar() }) {
```

Comment:
In this PR, the Scaffold includes a floatingActionButton argument, which is a button and can be defined as needed. This approach could lead to a more structured implementation.

Reply:
i had specifically avoided it to prevent the button from floating over the ingredients, as the ingredients are clickable to be expanded and it may be a pain if they're layered on top of each other. But I don't mind it either way

Reply:
Ok I see !!